schematools/contrib/django/factories.py

```diff
@@ -27,6 +27,212 @@
 TypeAndSignature = Tuple[Type[models.Field], tuple, Dict[str, Any]]
 
 
+class RelationMaker:
+    """ Superclass to generate info for relation fields """
+
+    def __init__(
+        self,
+        dataset: DatasetSchema,
+        table: DatasetTableSchema,
+        field: DatasetFieldSchema,
+        field_cls,
+        *args,
+        **kwargs,
+    ):
+        self.dataset = dataset
+        self.table = table
+        self.field = field
+        self._field_cls = field_cls
+        self._args = args
+        self._kwargs = kwargs
+        self.relation = field.relation or field.nm_relation
+        self.fk_relation = field.relation
+        self.nm_relation = field.nm_relation
+
+    @classmethod
+    def fetch_relation_parts(cls, relation):
+        relation_parts = relation.split(":")
+        return relation_parts[:2]
```

Comment:
I am not yet familiar with this code, hence I have no idea what `relation_parts[:2]` means. You could make the code somewhat self documenting by:

```python
a, b, c, *_ = relation.split(":")
return [a, b, c]
```
Where `a`, `b` and `c` have meaningful names that describe what they are.

Alternatively add a docstring describing what parts of the relation are actually being returned.

Reply:
Good point, changed it in `dataset_id`, `table_id`.